src/Middleware/RateLimiting/src/RateLimiterOptions.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Threading.RateLimiting;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Specifies options for the rate limiting middleware.
+/// </summary>
+public sealed class RateLimiterOptions
+{
+    // TODO - Provide a default?
+    private PartitionedRateLimiter<HttpContext> _limiter = new NoLimiter<HttpContext>();
+    private Func<HttpContext, RateLimitLease, Task> _onRejected = (context, lease) =>
+    {
+        return Task.CompletedTask;
+    };
+    private int _defaultOnRejectionStatusCode = (int)HttpStatusCode.ServiceUnavailable;
+
+    /// <summary>
+    /// Gets or sets the <see cref="PartitionedRateLimiter{TResource}"/>
+    /// </summary>
+    public PartitionedRateLimiter<HttpContext> Limiter
+    {
+        get => _limiter;
+        set => _limiter = value ?? throw new ArgumentNullException(nameof(value));
+    }
+
+    /// <summary>
+    /// Gets or sets a <see cref="Func{HttpContext, RateLimitLease, Task}"/> that handles requests rejected by this middleware.
+    /// </summary>
+    public Func<HttpContext, RateLimitLease, Task> OnRejected
+    {
+        get => _onRejected;
+        set => _onRejected = value ?? throw new ArgumentNullException(nameof(value));
+    }
+
+    /// <summary>
+    /// Gets or sets the default <see cref="HttpStatusCode"/> to set on the response when a request is rejected.
+    /// Defaults to <see cref="HttpStatusCode.ServiceUnavailable"/>.
+    /// </summary>
+    /// <remarks>
+    /// This status code will be set before <see cref="OnRejected"/> is called, so any status code set by
+    /// <see cref="OnRejected"/> will "win" over this default.
+    /// </remarks>
+    public int DefaultRejectionStatusCode
+    {
+        get => _defaultOnRejectionStatusCode;
+        set
+        {
+            if (!Enum.IsDefined(typeof(HttpStatusCode), value))
```

Comment:
CC @Tratcher @BrennanConroy @halter73 @davidfowl here's where I check the value of the default rejection status code

Reply:
This seems overly defensive to me. I'm pretty sure Kestrel will let you write any status code your heart desires.

Reply:
Removed the check